src/main/java/org/spongepowered/api/command/spec/CommandSpec.java

```diff
@@ -354,6 +354,15 @@ public boolean testPermission(CommandSource source) {
     }
 
     /**
+     * Gets the extended description used with this command if any is present
+     *
+     * @return the extended description.
+     */
+    public Optional<Text> getExtendedDescription(CommandSource source) {
```

Comment:
Well we could change `extendedDescription` to a `Function<CommandSource , Text>` to show some target specific help.
This would allow the help to show user specific limits or hide components the user does not have the permission to use.